build-scripts/compile_profiles.py

```diff
@@ -15,17 +16,30 @@ def __init__(self, * args, ** kwargs):
 
     def resolve(self, all_profiles):
         if self.resolved:
-            return self.raw_selections
+            return
         if self.extends:
+            if self.extends not in all_profiles:
+                msg = (
+                    "Profile {name} extends profile {extended}, but"
+                    "only profiles {known_profiles} are available for resolution."
+                    .format(name=self.id_, extended=self.extends,
+                            profiles=list(all_profiles.keys())))
+                raise RuntimeError(msg)
             extended_profile = all_profiles[self.extends]
-            extended_selects = extended_profile.resolve(all_profiles)
+            extended_profile.resolve(all_profiles)
 
+            extended_selects = extended_profile.raw_selections
             self.raw_selections.update(extended_selects)
-            self.variables.update(extended_profile.variables)
+
+            updated_variables = dict(extended_profile.variables)
+            updated_variables.update(self.variables)
+            self.variables = updated_variables
+
         self.raw_selections.update(set(self.selected))
```

Comment:
To keep related lines close, move line above after `self.raw_selections.update(extended_selects)`.